enterprise_subsidy/apps/api/v1/views.py

```diff
@@ -1 +1,216 @@
-# Create your views here.
+"""
+Views for the enterprise-subsidy service.
+"""
+from edx_rbac.mixins import PermissionRequiredForListingMixin
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from openedx_ledger.models import Transaction
+from rest_framework import permissions, status, viewsets
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.response import Response
+
+from enterprise_subsidy.apps.api.v1 import utils
+from enterprise_subsidy.apps.api.v1.serializers import SubsidySerializer, TransactionSerializer
+from enterprise_subsidy.apps.subsidy.constants import (
+    ENTERPRISE_SUBSIDY_ADMIN_ROLE,
+    ENTERPRISE_SUBSIDY_LEARNER_ROLE,
+    ENTERPRISE_SUBSIDY_OPERATOR_ROLE,
+    PERMISSION_CAN_CREATE_TRANSACTIONS,
+    PERMISSION_CAN_READ_SUBSIDIES,
+    PERMISSION_CAN_READ_TRANSACTIONS,
+    PERMISSION_NOT_GRANTED
+)
+from enterprise_subsidy.apps.subsidy.models import EnterpriseSubsidyRoleAssignment, Subsidy
+
+
+class SubsidyViewSet(PermissionRequiredForListingMixin, viewsets.ModelViewSet):
+    """
+    ViewSet for the Subsidy model.
+    """
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    lookup_field = 'uuid'
+    serializer_class = SubsidySerializer
+
+    # Fields that control permissions for 'list' actions, required by PermissionRequiredForListingMixin.
+    list_lookup_field = 'enterprise_customer_uuid'
+    allowed_roles = [ENTERPRISE_SUBSIDY_ADMIN_ROLE, ENTERPRISE_SUBSIDY_OPERATOR_ROLE]
+    role_assignment_class = EnterpriseSubsidyRoleAssignment
+
+    def get_permission_required(self):
+        """
+        Return permissions required for the current requested action.
+
+        We override the function here instead of just setting the ``permission_required`` class attribute because that
+        attribute only supports requiring a single permission for the entire viewset.  This override logic allows for
+        the permission required to be based conditionally on the type of action.
+        """
+        permission_for_action = {
+            "list": PERMISSION_CAN_READ_SUBSIDIES,
+            "retrieve": PERMISSION_CAN_READ_SUBSIDIES,
+            "create": PERMISSION_NOT_GRANTED,
+            "update": PERMISSION_NOT_GRANTED,
+            "partial_update": PERMISSION_NOT_GRANTED,
+            "destroy": PERMISSION_NOT_GRANTED,
+        }
+        permission_required = permission_for_action.get(self.request_action, PERMISSION_NOT_GRANTED)
+        return permission_required
```

Comment:
This is super clean and helpful; there's probably a pattern we can extract from this and add to `edx_rbac.mixins`